utilities/backup/backup_engine.cc

```diff
@@ -1857,7 +1858,7 @@ IOStatus BackupEngineImpl::RestoreDBFromBackup(
         GetAbsolutePath(file), dst, Temperature::kUnknown /* src_temp */,
         file_info->temp, "" /* contents */, backup_env_, db_env_,
         EnvOptions() /* src_env_options */, options_.sync,
-        options_.restore_rate_limiter.get(), 0 /* size_limit */,
+        options_.restore_rate_limiter.get(), file_info->size,
```

Comment:
This seems fine with me.

Reply:
I've kept this change so that no change is needed for the current unit test (`BackupEngineRateLimitingTestWithParam.RateLimitingChargeReadInRestore`) and at least there's no rate limiter overcharge on the restore side.

Reply:
I reached out to @pdillinger on his opinion on change like this. The change is fine here but he pointed out "there's a pre-existing missed opportunity for sanity checking. If I'm reading the code correctly, it will only read up to size_limit but does not check (in applicable cases - almost all) that the file size is as expected at read time. I will create a techdebt task for this.".


Feel free to reach out to him on the techdebt task if you are interested and available (you don't have to if you have other stuff lined up)

Reply:
That make sense, thanks for checking!